pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/OpChainSchedulerService.java

```diff
@@ -48,20 +48,16 @@ public class OpChainSchedulerService extends AbstractExecutionThreadService {
    * Default cancel signal retention, this should be set to several times larger than
    * {@link org.apache.pinot.query.service.QueryConfig#DEFAULT_SCHEDULER_RELEASE_TIMEOUT_MS}.
    */
-  private static final long DEFAULT_SCHEDULER_CANCELLATION_SIGNAL_RETENTION_MS = 60_000L;
+  private static final long SCHEDULER_CANCELLATION_SIGNAL_RETENTION_MS = 60_000L;
 
   private final OpChainScheduler _scheduler;
   private final ExecutorService _workerPool;
-  private final Cache<Long, Long> _cancelledRequests;
+  private final Cache<Long, Long> _cancelledRequests = CacheBuilder.newBuilder()
```

Comment:
Shall it still be tunable?

Reply:
shouldn't be a problem. cancellation signal are bounded by the control flow not the mailbox data flow, which has a much smaller timeout

Reply:
Hmm, what's the behavior when you have a very long timeout query? Or we should limit it from both control/data flows